File: PA28.java
Leetcode: 1. Two Sum
//  Time Complexity : O(n) // n is number of elements in nums
//Space complexity : O(n)   // n is number of elements in nums
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums== null ){
            return new int[0];
        }
        HashMap<Integer,Integer> hm =new HashMap<>();
        int[] res = new int[2];
        for(int i=0;i< nums.length;i++){
            
            if(hm.containsKey(target-nums[i])){
                res[1]= i;
                res[0]= hm.get(target-nums[i]);
                return res;
            }
            else{
                hm.put(nums[i],i);
            }
        }
        return new int[0];
        
    }
}

File: PA_28.py
#LC 1 - Two Sum
#Time Complexity - O(n)
#Space Complexity - O(n)
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()
        for i in range(len(nums)):
            if nums[i] not in d.keys():
                d[target-nums[i]] = i
            elif nums[i] in d.keys():
                if i != d[nums[i]]:
                    return [d[nums[i]],i]
                